pkg/controller/machine/reconcile_test.go

```diff
@@ -93,18 +96,31 @@ func TestMachineSetControllerReconcileHandler(t *testing.T) {
 			withFinalizer:  false,
 		},
 		{
-			name:           "Delete machine, skip master",
-			objExists:      true,
-			instanceExists: true,
-			isDeleting:     true,
-			withFinalizer:  true,
-			isMaster:       true,
+			name:                   "Delete machine, controller on different node with same name, but different UID",
+			objExists:              true,
+			instanceExists:         true,
+			isDeleting:             true,
+			withFinalizer:          true,
+			numExpectedDeleteCalls: 1,
+			expectFinalizerRemoved: true,
+			nodeName:               "controller-node-name",
+			nodeUID:                "another-uid",
+		},
+		{
+			name:                   "Delete machine, controller on the same node",
```

Comment:
I feel like there is at least one test missing. Maybe a specific set of tests for isDeleteAllowed would make sense as that function has a bunch of branches.

Reply:
Yeah, I agree, i'll add it.